### DISCUSSION TITLE
Export timesheets including absence days

### DISCUSSION THREAD

**Discussion by pkoevesdi** [kimai/kimai#5207] [Questions & Answers]
Before I spend hours to figure it out the hard way:
Is it possible to export timesheets having worked hours per day and absence days for every single user, across all customers?
We need something like this per user per month:

User Joe |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |  
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
day of month | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14 | 15 | 16 | 17 | 18 | 19 | 20 | 21 | 22 | 23 | 24 | 25 | 26 | 27 | 28 | 29 | 30 | 31
worked hours |   |   | 8 | 4 | 8 |   |   | 8 | 8 | 8 | 8 | 8 |   |   | 8 | 8 | 8 | 8 | 8 |   |   | 8 | 8 | 8 | 8 | 8 |   |   | 8 | 8 | 8
absence | 8 | 8 |   | 4 |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |  

If it is possible, where should I start? I assume, with "export", not "invoice", because ["Invoices can only be created for a dedicated customer, where an export can be created without selecting a customer"](https://www.kimai.org/documentation/export.html)
Which template file might be the best to start with?

Since we have to work with the exported data, we'd need excel at best, but html would also be ok.

**Comment by kevinpapst** [comment]
That is not possible without writing your own plugin and asking the internal APIs.

**Reply by kevinpapst** [reply]
But please send me full proposals, as I need to understand the different needs of companies.

**Reply by pkoevesdi** [reply]
Thank You!

We (and many other companies) have a public funding. To get the funding we must account the worked hours per month and per employee. This is the form:
https://www.ptj.de/lw_resource/datapool/systemfiles/elements/files/DC884FF9F1384510E0537E695E86CECD/live/document/Stundennachweis_NKBF_2017.xlsx

As a first step, we must not fill in directly from kimai, we can copy/paste it easily. But we need the data summed up by day and user. The crucial point atm is the absence part of it, how can we export these together with the worked hours?

Quite close is the view `/de/contract` already, we can copy from there. What's missing in the copied data is the differentiation between absence hours and worked hours, and we'd need the worked hours directly, not the plus/minus. But the latter can be worked around by temporarily setting the Sollstunden to zero.

**Comment by pkoevesdi** [comment]
With the new working contract plugin version 1.16.0, this comes quite near to what we need. It's only a pdf, with some extra work to paste it into excel, but it's already a very good step into that direction. Thank You!

**Reply by kevinpapst** [reply]
You were quick with noticing that. Glad it is a first step into the right direction. The approval workflow will likely keep me busy for a while, but I will keep on improving this.
That's why such qualified feedback is so important, thanks 👍

**Comment by pkoevesdi** [comment]
I got the information now from our funding people, that also the type of absence would be needed. We have a strong demand and must go into programming something ourselves. Before that I want to ask: Is there some chance that this feature request is in work already by Kimai?

**Reply by kevinpapst** [reply]
Planned, but not already in work

**Reply by pkoevesdi** [reply]
Thank You, great news. Any rough estimation on the implementation? More a question of weeks or years? ;-)
